src/fragment-components/a-icons.tsx

```diff
@@ -0,0 +1,184 @@
+				size: [{ size: size, text: sizeItems.find((sizeItem: any) => sizeItem.id === size)?.text, component: item[1] }],
+				key: icon,
+				label: `${icon}`,
+				name: item[1].render.name,
+				type: 'icons',
+				className: elementTileStyle,
+				selectedIcon: item[1],
+				selectedSize: size
+			}
+		};
+		if(item[0] !== 'Icon') {
+			const isIncluded = items.some((item: any) => item.key === object.key);
+			if(isIncluded) {
+				const current = items.find((item: any) => item.key === object.key);
+				current.componentObj.size.push({ size: size,
+					text: sizeItems.find((sizeItem: any) => sizeItem.id === size)?.text, component: item[1] });
+			} else {
+				items.push(object);
+			}
+		}
+	});
+	return items;
+};
+
+export const AIconsSettingsUI = ({ selectedComponent, setComponent }: any) => {
+	const [filterString, setFilterString] = useState('');
+	const shouldShow = (matches: string[]) => {
+		return !filterString || matches.some((match) => match.includes(filterString));
+	};
+	return (<>
+        <Dropdown
+        label='Size'
+        titleText='Size'
+        items={selectedComponent.size}
+        initialSelectedItem={selectedComponent.size.find((item: any) => item.size === selectedComponent.size[0].size)}
+        itemToString={(item: any) => (item ? item.text : '')}
+        onChange={(event: any) => setComponent({
+                ...selectedComponent,
+                selectedIcon: selectedComponent.size.find((item: any) => item.size === event.selectedItem.size).component,
+                selectedSize: selectedComponent.size.find((item: any) => item.size === event.selectedItem.size).size
+        })}/>
+		<Search id='icons-search'
+			className={searchStyle}
+			light
+			labelText='Filter icons'
+			placeholder='Filter icons'
+			onChange={(event: any) => setFilterString(event.target.value)} />
+        <div className={elementTileListStyle}>
+            {
+                 selectedComponent.items.filter((component: any) => shouldShow(component.componentObj.keywords)).map((props: any) => {
+                    const Component = props.componentObj.selectedIcon;
+                   // eslint-disable-next-line react/jsx-key
+                   return (<ElementTile componentObj={props.componentObj}>
+                                <Component
+                                onClick={() => setComponent({
+                                        ...selectedComponent,
+                                        selectedIcon: props.componentObj.selectedIcon,
+                                        key: props.componentObj.key,
+                                        size: props.componentObj.size,
+                                        name: props.componentObj.name
+                                    })}>
+                                </Component>
+                            </ElementTile>);
+                 })
+            }
+        </div>
+	</>);
+};
+export const AIcons = ({
```

Comment:
Isn't this a single icon component, but you can select out of all the available icons?

File name should change in that case too.